### PR TITLE
README updated

### PR DESCRIPTION
**Pull request by carocad**
fix: missing spec @mehdisadeghi I found out about this today when [trying](http://try.hiposfer.com/index.html) our server

I also just created a wiki entry for routing: https://github.com/hiposfer/service.routing/wiki/Routing-Overview

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,52 +1,65 @@
...
-FIXME: explanation
+    $ docker run -it -p 3000:3000 hiposfer/service.routing
 
-[//]: # ($ java -jar sagacious-woof-0.1.0-standalone.jar [args])
 
+### Development
 
-## Options
+We follow [Stuart Sierra's reload workflow](https://github.com/stuartsierra/component).
+The system configuration is created [here](https://github.com/hiposfer/service.routing/blob/master/src/hiposfer/service/routing/core.clj#L77)
```
Comment (mehdisadeghi) [inline README.md]: I think the details belong to the wiki.

File: README.md [hunk 2]
```diff
@@ -1,52 +1,65 @@
...
+by merging the [default config](https://github.com/hiposfer/service.routing/blob/master/resources/default-config.edn)
+with the [required environmental variables](https://github.com/hiposfer/service.routing/blob/master/src/hiposfer/service/routing/core.clj#L84).
```
Comment (mehdisadeghi) [inline README.md]: We better avoid pointing to source code lines in README. Who is going to take care of updating these links after all!? :)

**Comment by carocad** [comment]
@mehdisadeghi not related to this PR but I think generally relevant.

What do you think of changing the naming convention as follows:
- app(lications) should have simple names. For example: `routing` instead of `service.routing`, `hive` instead of `app.hive`
- libraries should have compound names based on what they do. For example: `math.frechet`, `geojson.specs`. This is already like that so no need to change anything here, just reasserting it :)

I want to bring up this topic because now that the org name is part of the namespace, the imports become really long without any extra meaning. For example: `hiposfer.service.routing.graph.core`.
`service` doesnt provide any extra meaning here (imo), but rather just adds noise.

I think it would also help to organize the code like this since app(lications) are generally top-level compositions of smaller parts, but currently our apps dont give that impression.

What do you think?

File: README.md (hunk 2 above)
Comment (carocad) [reply README.md]: Good point, I will change this.

**Comment by mehdisadeghi** [comment]
@carocad 

> app(lications) should have simple names. For example: routing instead of service.routing, hive instead of app.hive

I totally agree. It is lean and mean. I also think it is a good idea to choose relatively short and catchy names for applications regardless of their functionality.

> libraries should have compound names based on what they do. For example: math.frechet, geojson.specs.

I almost agree. IMO, we have to prefer names that makes it easier to _communicate_ with other people, i.e. names that are easier to pronounce and refer to, both in oral and written communication forms.

**Comment by carocad** [comment]
@mehdisadeghi thanks for your input ! :)

> I totally agree. It is lean and mean. I also think it is a good idea to choose relatively short and catchy names for applications regardless of their functionality

Agree. If you know a good name for it, please by all means propose it. I think we are both very bad at naming things so I just went with the easiest one :D

> I almost agree. IMO, we have to prefer names that make it easier to communicate with other people, i.e. names that are easier to pronounce and refer to, both in oral and written communication forms.

I partially agree on this: Having short and easy names is definitely a good thing. Nevertheless, I think that somehow this has gotten a bit out of control in the software industry where every little thing gets a new "shinny" name that tells the user nothing about it.

I generally try to approach this as follows: If I look into a project dependencies (regardless of the language), would I understand its technology stack just with that information?

We could for example rename `geojson.specs` as `thunderbolt` or something like that. But that tells the developers that use it, nothing about it. Neither as a dependency, nor as a namespace import.

I think that, just as you said, being able to communicate and refer to something in an easy way is definitely something that we should strive for. That is the reason why, so far, I have used `boring` but easy to understand names.

If you want to read more about this "philosophy" here is a link with more info . Hope you like it :)
https://gitlab.com/gitlab-com/blog-posts/issues/31

**Comment by carocad** [comment]
@mehdisadeghi What do you think of:
- Sextant
- Kamal

as possible names for this app? 

I pulled them out of here: https://www.wikiwand.com/en/Navigational_instrument :D

**Comment by mehdisadeghi** [comment]
> What do you think of: sextant kamal

I have no strong opinion on kamal but I would definitely not go by _sextant_! :D

**Pull request merged**